src/commands/scaling/addScaleRule/GetScaleRuleNameStep.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ContainerApp, ContainerAppsAPIClient, ScaleRule } from '@azure/arm-appcontainers';
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import { createContainerAppsAPIClient } from '../../../utils/azureClients';
+import { getResourceGroupFromId } from '../../../utils/azureUtils';
+import { localize } from '../../../utils/localize';
+import { IAddScaleRuleWizardContext } from './IAddScaleRuleWizardContext';
+
+export class GetScaleRuleNameStep extends AzureWizardPromptStep<IAddScaleRuleWizardContext> {
+    public hideStepCount: boolean = true;
+    containerApp: ContainerApp | undefined;
+    scaleRules: ScaleRule[] | undefined;
+
+    public async prompt(context: IAddScaleRuleWizardContext): Promise<void> {
+        context.ruleName = (await context.ui.showInputBox({
+            prompt: localize('scaleRuleNamePrompt', 'Enter a name for the new scale rule.'),
+            validateInput: async (value: string | undefined): Promise<string | undefined> => await this.validateInput(context, value)
+        })).trim();
+    }
+
+    public shouldPrompt(): boolean {
+        return true;
+    }
+
+    private async initContainerAppClient(context: IAddScaleRuleWizardContext): Promise<void> {
+        try {
+            const client: ContainerAppsAPIClient = await createContainerAppsAPIClient([context, context.treeItem]);
+            const containerAppRg: string = getResourceGroupFromId(context.treeItem.fullId);
+            this.containerApp = await client.containerApps.get(containerAppRg, context.treeItem.parent.parent.label);
```

Comment:
You should already have this data (it lives in the `ContainerAppTreeItem`) so you shouldn't have to create a new client and get this from Azure.